src/hoc/DragDropHandler.tsx

```diff
@@ -1,58 +1,96 @@
 'use client'
 
-import { ReactNode, useRef } from 'react'
+import { selectTaskBoard } from '@/redux/features/taskBoardSlice'
+import { TaskResponse } from '@/types/dto/tasks.dto'
+import { View } from '@/types/interfaces'
+import { ViewMode } from '@prisma/client'
+import { ReactNode, useEffect, useRef } from 'react'
 import { useDrag, useDrop, DropTargetMonitor } from 'react-dnd'
+import { useSelector } from 'react-redux'
 
 interface Prop {
   children: ReactNode
   accept: string
   index: number
-  id: string
+  id?: string //only pass for droppable
+  task?: TaskResponse //only pass for draggable
   moveCard?: (dragIndex: number, hoverIndex: number, sourceId: number, targetId: number) => void
   onDropItem?: (payload: { taskId: string; targetWorkflowStateId: string }) => void
-  draggable?: boolean // Add this prop
+  draggable?: boolean // Indicates if the item should be draggable
+  droppable?: boolean // Indicates if the item should be droppable
 }
 
-export const DragDropHandler = ({ children, accept, index, id, moveCard, onDropItem, draggable = false }: Prop) => {
+export const DragDropHandler = ({
+  children,
+  accept,
+  index,
+  id,
+  task,
+  moveCard,
+  onDropItem,
+  draggable = false,
+  droppable = false, // New prop for droppable
+}: Prop) => {
+  const { view } = useSelector(selectTaskBoard)
   const ref = useRef<HTMLDivElement | null>(null)
 
-  const [, drop] = useDrop({
+  const [{ isOver, canDrop }, drop] = useDrop({
```

Comment:
Can we do this instead? or something like this. Try to not use typecasting if possible.

const [{ isOver, canDrop }, drop] = useDrop<{ task: TaskResponse }, unknown, { isOver: boolean; canDrop: boolean }>({
    accept: accept,
    collect: (monitor: DropTargetMonitor) => ({
      isOver: monitor.isOver(),
      canDrop: monitor.canDrop(),
    }),
    drop: (item, monitor) => {
      if (onDropItem) {
        onDropItem({
          taskId: item.task.id,
          targetWorkflowStateId: id as string,
        })
      }
    },
    canDrop: () => droppable, // Only allow dropping if the component is droppable
  })

Reply:
Fixed in the commit below